### PR TITLE
Added all editable homepage regions

### LINKED ISSUE #64: Make home page content editable
# Overview

The client would like to edit the contents of the homepage directly on the site via an admin interface. CKAN was not designed for this use case, so we will need to decide what to make editable and how.

https://i.imgur.com/OXMqKzD.jpg

The work can be done in different stages, for the first iteration we will provide a basic interface for changing the texts of the home page and perhaps the featured image and then we can explore ways of improving it.

Here's an overview of how it could work:

* For each of the text snippets in the home page and the featured image there will be a config option (we'll need to think of a sane naming convention)
* CKAN config options can be made editable at runtime ([docs](http://docs.ckan.org/en/2.8/extensions/remote-config-update.html)), and they can be added to the admin UI under `/ckan-admin/` in a new tab. These config options are then stored in the database.
* The templates will be updated to use a helper that will retrieve the relevant text, eg:

```html

<h2>What's happening</h2>
<p>{{ h.get_text_snippet('ckan.lacounts.text.whats_happening_intro') }} 
```

----

# Additional tasks

> decline, implement or create new issues

- [x] implement image uploading - http://getcontenttools.com/tutorials/handling-image-uploads? - MOVED #77
- [x] create all required regions in templates
- [x] update the corresponding validator
- [x] create `macros/editable.html`

### PR DESCRIPTION
**Pull request by roll**
- fixes #64

### CODE REVIEW

File: ckanext/lacounts/templates/home/snippets/community.html [hunk 1]
```diff
@@ -1,8 +1,15 @@
+{# TODO: It doesn't work inside `macros/editable.html' #}
```
Comment (roll) [inline ckanext/lacounts/templates/home/snippets/community.html]: @amercader 
I get brutal `internal error` on any attempt of moving this duplicated code to a `macros` or `snippets` file.

**Comment by roll** [comment]
@amercader 
can you please review? Also, can you please check it works for you. I can't reproduce the error you mentioned yesterday.

File: ckanext/lacounts/templates/home/snippets/promoted.html [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-        <h3><a href="#"><small>{{ _("Featured visualisation") }}:</small> Where are L.A.'s dirtiest streets?</a></h3>
+        <div><a href="#"><small>{{ _("Featured visualisation") }}:</small></a></div>
         <button type="button" class="toggle"><span class="sr-only">Toggle description</span></button>
       </div>
       <div class="caption-body in">
-        <p>
-          Inspectors gave failing cleanliness grades to more than 370 miles of streets and alleys in Los Angeles, or about 4% of the city. The grades were based on observations of illegal dumping, bulky items, loose litter and weeds in those areas.
-        </p>
+        {% call editable.region('featured-visualisation') %}
+          <h3>Where are L.A.'s dirtiest streets?</h3>
```
[the remark this answers is not in the document]
Comment (smth) [reply ckanext/lacounts/templates/home/snippets/promoted.html]: All sounds reasonable to me.

**Comment by roll** [comment]
Thanks @amercader @smth

**Pull request merged**